Model/Adapter/PurchaseVerify.php

```diff
@@ -217,22 +222,37 @@ private function _handleCustomerEmail($quote) {
             return;
         }
 
-        if (!empty($quote->getShippingAddress()->getEmail())) {
-            $quote->setCustomerEmail((string)$quote->getShippingAddress()->getEmail());
+        if (!empty($quote->getBillingAddress()) && !empty($quote->getBillingAddress()->getEmail())) {
+            $quote->setCustomerEmail((string)$quote->getBillingAddress()->getEmail());
             return;
         }
 
-        if (!empty($quote->getBillingAddress()->getEmail())) {
-            $quote->setCustomerEmail((string)$quote->getBillingAddress()->getEmail());
+        if (!empty($quote->getShippingAddress()) && !empty($quote->getShippingAddress()->getEmail())) {
+            $quote->setCustomerEmail((string)$quote->getShippingAddress()->getEmail());
             return;
         }
     }
 
+    private function _standardizeAddress($quote){
+        $billing = $quote->getBillingAddress();
+        $shipping = $quote->isVirtual() ? null : $quote->getShippingAddress();
+
+        //if not virtual but shipping email empty, use billing address email info on shipping
+        if ($shipping && empty($shipping->getEmail())) {
+            $this->logger->info('Shipping address email empty, copying from billing: '.$billing->getEmail());
+            $shipping->setEmail($billing->getEmail());
+            $quote->setShippingAddress($shipping);
```

Comment:
Should this be 
`$quote->setShippingAddress($billing)`

?

Reply:
this one is correct, I updated the shipping object's email on line 243, so I want to set it back to the quote just to be on the safe side